src/core/Microsoft.PowerToys.Settings.UI.Lib/SettingsUtils.cs

```diff
@@ -59,16 +60,53 @@ public bool SettingsExists(string powertoy = DefaultModuleName, string fileName
 
         /// <summary>
         /// Get a Deserialized object of the json settings string.
+        /// This function creates a file in the powertoy folder if it does not exist and returns an object with default properties.
         /// </summary>
         /// <returns>Deserialized json settings object.</returns>
         public T GetSettings<T>(string powertoy = DefaultModuleName, string fileName = DefaultFileName)
+            where T : ISettingsConfig, new()
+        {
+            if (SettingsExists(powertoy, fileName))
+            {
+                var deserializedSettings = GetFile<T>(powertoy, fileName);
+
+                // If GeneralSettings is being accessed for the first time, perform a check on the version number and update accordingly.
+                if (powertoy.Equals(DefaultModuleName))
+                {
+                    try
+                    {
+                        if (Helper.CompareVersions(((GeneralSettings)(object)deserializedSettings).PowertoysVersion, Helper.GetProductVersion()) < 0)
+                        {
+                            // Update settings
+                            ((GeneralSettings)(object)deserializedSettings).PowertoysVersion = Helper.GetProductVersion();
+                            SaveSettings(((GeneralSettings)(object)deserializedSettings).ToJsonString(), powertoy, fileName);
+                        }
+                    }
+                    catch (FormatException)
+                    {
+                        // If there is an issue with the version number format, don't migrate settings.
+                    }
+                }
+
+                return deserializedSettings;
+            }
+            else
+            {
+                // If the settings file does not exist, to create a new object with default parameters and save it to a newly created settings file.
+                T newSettingsItem = new T();
+                SaveSettings(newSettingsItem.ToJsonString(), powertoy, fileName);
+                return newSettingsItem;
+            }
+        }
+
+        // Given the powerToy folder name and filename to be accessed, this function deserializes and returns the file.
+        public T GetFile<T>(string powertoyFolderName = DefaultModuleName, string fileName = DefaultFileName)
```

Comment:
I'm confused why this needs to be public.  What happens in the cases that the file doesn't exist when it's called from powerlauncher / keyboard manager etc?

Reply:
In the case of PT Run, it catches the exception and runner creates the file. In the case of KBM, it catches the exception and KBM, out of runner creates the file. One thing that I can do is to check if the file exists and only then call getSettings, else do whatever is being done in the catch block.

Reply:
Yes, for KBM I think we don't really need a try-catch, it could just be a file exists check since that is the only expected scenario where the file read would fail. There is a named mutex on the file as well to ensure both KBM from runner and Settings don't access that file simultaneously.

Reply:
Made it private.